### PR TITLE
Add support for numpy 2.0.0 

### LINKED ISSUE #386: Add support for numpy 2.0.0
### Problem Description
- Currently, we restrict numpy < 2 for all python versions.
- We would like to support the newest major release of numpy. 

### Expected behavior
- CTGan supports numpy 2.0

### Additional context
- Remove upper bound restriction in pyproject.toml
- Update any associated tests

### PR DESCRIPTION
**Pull request by R-Palazzo**
Resolve #386
CU-86b0zmjw0

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b0zmjw0 CTGAN - Add support for numpy 2.0.0  #386](https://app.clickup.com/t/86b0zmjw0)

File: pyproject.toml [hunk 1]
```diff
@@ -179,7 +179,9 @@ select = [
     # print statements
     "T201",
     # pandas-vet
-    "PD"
+    "PD",
+    # numpy 2.0
+    "NPY201"
```
Comment (amontanez24) [inline pyproject.toml:184]: what is this?

Comment (gsheni) [reply pyproject.toml:184]: - Let's `ruff` check for old numpy usage:
  - https://docs.astral.sh/ruff/rules/numpy2-deprecation/

Comment (R-Palazzo) [reply pyproject.toml:184]: Hi @amontanez24, exactly; on CTGAN, there were no changes (no new linting errors), but I found 6 NPY201 linting errors on SDV. Let me know if we should keep it here.

Comment (amontanez24) [reply pyproject.toml:184]: Ok, we can keep

**Review by amontanez24 — APPROVED** [review]

**Review by fealho — APPROVED** [review]

**Pull request merged**